modules/core/src/main/java/org/apache/ignite/internal/managers/communication/GridIoManager.java

```diff
@@ -3289,8 +3289,15 @@ public void send(
             catch (IgniteCheckedException e) {
                 closeChannelQuiet();
 
-                throw new IgniteCheckedException("Exception while sending file [rmtId=" + rmtId +
-                    ", file=" + file.getName() + ", sesKey=" + sesKey + ", retries=" + retries + ']', e);
+                if (X.hasCause(e, TransmissionCancelException.class)) {
+                    throw new TransmissionCancelException("File transmission has been cancelled on the remote node " +
+                        "[rmtId=" + rmtId + ", file=" + file.getName() + ", sesKey=" + sesKey +
+                        ", retries=" + retries + ", cause='" + e.getCause().getMessage() + "']");
```

Comment:
It seems that `e.getCause()`may not always contain expected exception (TransmissionCancelException).